pkg/daemon/ceph/client/pool.go

```diff
@@ -418,6 +421,68 @@ func CreateReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterI
 	}
 
 	logger.Infof("creating replicated pool %s succeeded", poolName)
+
+	if checkFailureDomain {
+		if err = ensureFailureDomain(context, clusterInfo, clusterSpec, poolName, pool); err != nil {
+			return nil
+		}
+	}
+	return nil
+}
+
+func ensureFailureDomain(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, poolName string, pool cephv1.PoolSpec) error {
+	if pool.FailureDomain == "" {
+		logger.Debugf("skipping check for failure domain on pool %q as it is not specified", poolName)
+		return nil
+	}
+
+	logger.Debugf("checking that pool %q has the failure domain %q", poolName, pool.FailureDomain)
+	details, err := GetPoolDetails(context, clusterInfo, poolName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get pool %q details", poolName)
+	}
+
+	// Find the failure domain for the current crush rule
+	rule, err := getCrushRule(context, clusterInfo, details.CrushRule)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get crush rule %q", details.CrushRule)
+	}
+	currentFailureDomain := ""
+	if len(rule.Steps) < 2 || rule.Steps[1].Type == "" {
+		logger.Warningf("unexpected format for crush rule %q, proceeding to create a new crush rule", details.CrushRule)
+	} else if rule.Steps[1].Type != pool.FailureDomain {
```

Comment:
@leseb Any suggestions on detecting the crush rule failure domain other than this? This code path only applies to a basic replica pool, and not to EC, stretch, or pools with sub-failure domains.